### PR TITLE
chore: Rename core and labs/core packages to tokens and labs/tokens

### LINKED ISSUE #1006: Change core package to tokens
## 🚀 Feature Proposal

* Change the canvas-kit-core package to canvas-kit-tokens
* Change the canvas-kit-labs-core package to canvas-kit-labs-tokens
* Create a codemod to help consumers make this update

## Motivation

This removes ambiguity between core and common and aligns the package with tokens in our design language

### PR DESCRIPTION
**Pull request by alanbsmith**
## Summary

Resolves #1006 

• add initial codemod
• add initial codemod tests
• update package readmes

## Checklist

- [ ] branch has been rebased on the latest master commit
- [ ] tests are changed or added
- [ ] `yarn test` passes
- [ ] all (dev)dependencies that the module needs is added to its `package.json`
- [ ] code has been documented and, if applicable, usage described in README.md
- [ ] module has been added to `canvas-kit-react` and/or `canvas-kit-css` universal modules, if
      applicable
- [ ] design approved final implementation
- [ ] a11y approved final implementation
- [ ] code adheres to the [API & Pattern guidelines](../API_PATTERN_GUIDELINES.md)

### CODE REVIEW

File: modules/codemod/lib/v5/renameCoreImports.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+type SourceMap = {
+  '@workday/canvas-kit-react/core': '@workday/canvas-kit-react/tokens';
+  '@workday/canvas-kit-labs-react/core': '@workday/canvas-kit-labs-react/tokens';
+};
+
+const sourceMap: SourceMap = {
+  '@workday/canvas-kit-react/core': '@workday/canvas-kit-react/tokens',
+  '@workday/canvas-kit-labs-react/core': '@workday/canvas-kit-labs-react/tokens',
+};
```
Comment (NicholasBoll) [inline modules/codemod/lib/v5/renameCoreImports.ts:11]: What's the need to type this twice? Why not:

```suggestion

const sourceMap = {
  '@workday/canvas-kit-react/core': '@workday/canvas-kit-react/tokens',
  '@workday/canvas-kit-labs-react/core': '@workday/canvas-kit-labs-react/tokens',
} as const;
```

 Below you'd have to also change `as keyof SourceMap` to `as keyof typeof sourceMap`, but personally I prefer that. It reduces the Typescript noise.

File: modules/codemod/lib/v5/renameCoreImports.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  // transform labs core imports
+  root
+    .find(j.ImportDeclaration, {source: {value: '@workday/canvas-kit-labs-react/core'}})
+    .forEach(nodePath => {
+      const source = nodePath.value.source.value as keyof SourceMap;
+      nodePath.value.source.value = sourceMap[source];
+
+      return nodePath;
+    });
```
Comment (NicholasBoll) [inline modules/codemod/lib/v5/renameCoreImports.ts:37]: I guess a way to get around the `as keyof ...` business is to filter in the `forEach`. I wouldn't worry about which one is faster:

```suggestion
  root
    .find(j.ImportDeclaration)
    .forEach(nodePath => {
      const source = nodePath.value.source.value;
      if (source === '@workday/canvas-kit-labs-react/core') {
        nodePath.value.source.value = sourceMap[source];
      }

      return nodePath;
    });
```

File: modules/codemod/lib/v5/spec/renameCoreImports.spec.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      const input = `import {colors} from "@workday/canvas-kit-react/core";`;
+      const expected = `import {colors} from "@workday/canvas-kit-react/tokens";`;
+
+      expectTransform(input, expected);
+    });
+  });
+
+  context('when importing from canvas-kit-labs-react/core', () => {
+    // import type from '@workday/canvas-kit-labs-react-core
+    // import type, { type, colors, StaticStates } '@workday/canvas-kit-labs-react/core';
```
Comment (NicholasBoll) [inline modules/codemod/lib/v5/spec/renameCoreImports.spec.ts:19]: What is the purpose of these comments?

**Review by NicholasBoll — APPROVED** [review]
Looks like there is an error when rendering menu stories

File: modules/codemod/lib/v5/renameCoreImports.ts (hunk 1 above)
Comment (alanbsmith) [reply modules/codemod/lib/v5/renameCoreImports.ts:11]: Yeah, I could just do the one. 👍🏻

File: modules/codemod/lib/v5/spec/renameCoreImports.spec.ts (hunk 3 above)
Comment (alanbsmith) [reply modules/codemod/lib/v5/spec/renameCoreImports.spec.ts:19]: Oh, sorry. This is still a draft. Those are potential test cases.

File: modules/labs-react/select/stories/stories_visualTesting.tsx [hunk 4]
```diff
@@ -102,7 +102,7 @@ export const SelectStatesMenuOn = () => (
           >
             <SelectBase
               {...props}
-              buttonRef={buttonRef}
+              ref={buttonRef}
```
Comment (alanbsmith) [inline modules/labs-react/select/stories/stories_visualTesting.tsx:105]:
```suggestion
              buttonRef={buttonRef}
```

File: modules/react/button/stories/visual-testing/stories_IconButton.tsx [hunk 5]
```diff
@@ -28,23 +28,19 @@ export const IconButtonStates = (props: {theme?: PartialEmotionCanvasTheme}) =>
...
-                {value: "circleFilled", label: 'Circle Filled'},
-                {value: "square", label: 'Square'},
-                {value: "squareFilled", label: 'Square Filled'},
+                {value: 'inverse', label: 'Inverse'},
+                {value: 'inverseFilled', label: 'Inverse Filled'},
+                {value: 'plain', label: 'Plain'},
+                {value: 'circle', label: 'Circle'},
+                {value: 'circleFilled', label: 'Circle Filled'},
+                {value: 'square', label: 'Square'},
+                {value: 'squareFilled', label: 'Square Filled'},
```
Comment (alanbsmith) [inline modules/react/button/stories/visual-testing/stories_IconButton.tsx:37]: This was not a a part of the transform. 🤷🏻

File: modules/react/button/stories/visual-testing/stories_IconButton.tsx [hunk 6]
```diff
@@ -28,23 +28,19 @@ export const IconButtonStates = (props: {theme?: PartialEmotionCanvasTheme}) =>
...
             })}
             columnProps={stateTableColumnProps}
           >
             {props => (
-              <Container
-                blue={["inverse", "inverseFilled"].includes(
-                  props.variant
-                )}
-              >
+              <Container blue={['inverse', 'inverseFilled'].includes(props.variant)}>
```
Comment (alanbsmith) [inline modules/react/button/stories/visual-testing/stories_IconButton.tsx:43]: Nor was this. 🤷🏻

**Pull request merged**